Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,258 @@
+                textureNameID == MeshSyncConstants._MaskMap) {
+                mat.EnableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                mat.EnableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+
+                TextureImporter importer = (TextureImporter)AssetImporter.GetAtPath(savePath);
+                if (importer != null) {
+                    importer.sRGBTexture         = false;
+                    importer.alphaIsTransparency = false;
+                    importer.ignorePngGamma      = true;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to apply any pending material updates.
+        /// </summary>
+        private void UpdatePendingMaterials() {
+            for (int i = pendingMaterialUpdates.Count - 1; i >= 0; i--) {
+                Tuple<Material, int> pendingMaterial = pendingMaterialUpdates[i];
+                if (SetSerializedTextureForMaterial(pendingMaterial.Item1, pendingMaterial.Item2)) {
+                    pendingMaterialUpdates.RemoveAt(i);
+                }
+            }
+        }
+#endif
+
+        /// <summary>
+        /// Returns the default shader for the active render pipeline.
+        /// </summary>
+        private static Shader GetStandardShader() {
+#if AT_USE_HDRP
+            return Shader.Find("HDRP/Lit");
+#elif AT_USE_URP
+            return Shader.Find("Universal Render Pipeline/Lit");
+#else
+            return Shader.Find("Standard");
+#endif
+        }
+
+        /// <summary>
+        /// Sets the required shader for the given material.
+        /// </summary>
+        /// <param name="mat">Material to set the shader on</param>
+        /// <param name="shaderName">The name of the shader passed by the DCC tool</param>
+        private static void UpdateShader(ref Material mat, string shaderName) {
+            Shader shader = null;
+            if (!string.IsNullOrEmpty(shaderName)) {
+                shader = Shader.Find(shaderName);
+            }
+
+            bool shaderExists = shader != null;
+
+            if (shader == null) {
+                shader = GetStandardShader();
+            }
+
+            Assert.IsNotNull(shader);
+
+            if (mat == null) {
+                mat = new Material(shader);
+            }
+            else {
+                mat.shader = shader;
+            }
+
+            bool usingOverride = false;
+
+            // If the given shader name did not exist, try to set up the material to be close to the given shader name:
+            if (!shaderExists) {
+                if (shaderName.ToLower() == "glass") {
+                    SetupGlassShader(mat);
+                    usingOverride = true;
+                }
+            }
+
+            // If the material was set up by meshsync but doesn't need that anymore, revert it to a standard material:
+            if (!usingOverride && Array.IndexOf(mat.shaderKeywords, MeshSyncConstants.MESHSYNC_OVERRIDE) >= 0) {
```

Comment:
And `CreateDefaultMaterial()` also doesn't need to call this particular code, does it ?

Reply:
Not if it uses the `UpdateShader` method to handle creation and update in a single place.

Reply:
I saw your changes in `CreateDefaultMaterial()`.
Resolving this conversation.